pkg/k8shandler/visualization.go

```diff
@@ -283,17 +280,40 @@ func (clusterRequest *ClusterLoggingRequest) createOrUpdateKibanaDeployment(prox
 			current, different := isDeploymentDifferent(current, kibanaDeployment)
 
 			// Check trustedCA certs have been updated or not by comparing the hash values in annotation.
-			newTrustedCAHashedValue, err := calcTrustedCAHashValue(kibanaTrustBundle)
+			newTrustedCAHashValue, err := calcTrustedCAHashValue(kibanaTrustBundle)
 			if err != nil {
 				return fmt.Errorf("unable to calculate trusted CA value. E: %s", err.Error())
 			}
-			trustedCAHashedValue, _ := current.Spec.Template.ObjectMeta.Annotations[constants.TrustedCABundleHashName]
-			if trustedCAHashedValue != newTrustedCAHashedValue {
+			trustedCAHashValue, _ := current.Spec.Template.ObjectMeta.Annotations[constants.TrustedCABundleHashName]
+			if trustedCAHashValue != newTrustedCAHashValue {
 				different = true
 				if kibanaDeployment.Spec.Template.ObjectMeta.Annotations == nil {
 					kibanaDeployment.Spec.Template.ObjectMeta.Annotations = make(map[string]string)
 				}
-				kibanaDeployment.Spec.Template.ObjectMeta.Annotations[constants.TrustedCABundleHashName] = newTrustedCAHashedValue
+				kibanaDeployment.Spec.Template.ObjectMeta.Annotations[constants.TrustedCABundleHashName] = newTrustedCAHashValue
+			}
+
+			// Check secret hash has been updated or not
+			for _, secretName := range []string{"kibana", "kibana-proxy"} {
+
+				hashKey := fmt.Sprintf("%s%s", constants.SecretHashPrefix, secretName)
+
+				secret, err := clusterRequest.GetSecret(secretName)
+				if err != nil {
+
+				}
+				newSecretHashValue, err := calcSecretHashValue(secret)
+				if err != nil {
+
+				}
+				currentSecretHashValue, _ := current.Spec.Template.ObjectMeta.Annotations[hashKey]
```

Comment:
this can't be `nil` since its from the already existing object here which would just be empty -
https://godoc.org/k8s.io/apimachinery/pkg/apis/meta/v1#ObjectMeta

Reply:
sorry, golang weirdness.  technically it is nil (or at least ==nil returns true) but map["key"] does not get a nil ptr error, it just returns nothing.